src/interfaces/coral_web/src/components/Agents/AgentForm.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect } from 'react';
+
+import { CreateAgent } from '@/cohere-client';
+import { Checkbox, Dropdown, DropdownOptionGroups, Input, InputLabel } from '@/components/Shared';
+import { DEPLOYMENT_COHERE_PLATFORM, MODEL_COMMMAND_R_PLUS } from '@/constants';
+import { useListAllDeployments, useModels } from '@/hooks/deployments';
+import { useListTools } from '@/hooks/tools';
+import { cn } from '@/utils';
+
+export type AgentFormFields = Omit<CreateAgent, 'version' | 'temperature'>;
+export type AgentFormFieldKeys = keyof AgentFormFields;
+
+type Props = {
+  fields: AgentFormFields;
+  onChange: (key: Omit<AgentFormFieldKeys, 'tools'>, value: string) => void;
+  onToolToggle: (toolName: string, checked: boolean) => void;
+  errors?: Partial<Record<AgentFormFieldKeys, string>>;
+  className?: string;
+};
+/**
+ * @description Base form for creating/updating an agent.
+ */
+export const AgentForm: React.FC<Props> = ({
+  fields,
+  onChange,
+  onToolToggle,
+  errors,
+  className,
+}) => {
+  const { data: deployments } = useListAllDeployments();
+  const { models } = useModels(fields.deployment);
+  const deploymentOptions: DropdownOptionGroups = [
+    {
+      options: (deployments ?? []).map(({ name }) => ({
+        label: name,
+        value: name,
+      })),
+    },
+  ];
+  const modelOptions: DropdownOptionGroups = [
+    {
+      options: models.map((model) => ({
+        label: model,
+        value: model,
+      })),
+    },
+  ];
+  const { data: toolsData } = useListTools();
+  const tools = toolsData?.filter((t) => t.is_available) ?? [];
+
+  useEffect(() => {
```

Comment:
feel like its fine to have this side effect here since we always want this to happen